crypto/keyring/legacy_info.go

```diff
@@ -261,13 +241,33 @@ func unmarshalInfo(bz []byte) (info Info, err error) {
 	//
 	// This is a workaround, as go cannot check that an interface (Info)
 	// implements another interface (UnpackInterfacesMessage).
-	_, ok := info.(multiInfo)
+	_, ok := info.(LegacyMultiInfo)
 	if ok {
-		var multi multiInfo
+		var multi LegacyMultiInfo
 		err = legacy.Cdc.UnmarshalLengthPrefixed(bz, &multi)
 
 		return multi, err
 	}
 
 	return
 }
+
+// privKeyFromLegacyInfo exports a private key from LegacyInfo
+func privKeyFromLegacyInfo(info LegacyInfo) (cryptotypes.PrivKey, error) {
+
+	switch linfo := info.(type) {
+	case legacyLocalInfo:
+		if linfo.PrivKeyArmor == "" {
+			return nil, fmt.Errorf("private key not available")
+		}
+		priv, err := legacy.PrivKeyFromBytes([]byte(linfo.PrivKeyArmor))
+		if err != nil {
+			return nil, err
+		}
+
+		return priv, nil
+	// case legacyLedgerInfo, legacyOfflineInfo, LegacyMultiInfo:
```

Comment:
@robert-zaremba  I am thinking to set up `case legacyLedgerInfo, legacyOfflineInfo, LegacyMultiInfo`  and  cause panic at `default` case. Do you think it is a good idea?

Reply:
cc @robert-zaremba

Reply:
Error is better than panic.